File: ai-platform/frontend/lib/chains.ts
/**
 * HookSwap LIVE chain registry (mirrors ai-platform/README.md + the DEX
 * contracts/deployments/*.json). The AI platform filters retrieval by these
 * chains; new chains are added here by config alone.
 */
export interface ChainInfo {
  id: number;
  key: string;
  name: string;
  nativeCurrency: string;
  color: string;
  notes?: string;
  testnet?: boolean;
}

export const HOOKSWAP_CHAINS: ChainInfo[] = [
  { id: 4663, key: 'robinhood', name: 'Robinhood', nativeCurrency: 'ETH', color: '#00C805', notes: 'primary live chain' },
  { id: 999, key: 'hyperevm', name: 'HyperEVM', nativeCurrency: 'HYPE', color: '#97FCE4' },
  { id: 57073, key: 'ink', name: 'Ink', nativeCurrency: 'ETH', color: '#7A5CFA' },
  { id: 4326, key: 'megaeth', name: 'MegaETH', nativeCurrency: 'ETH', color: '#F5A623' },
  { id: 196, key: 'xlayer', name: 'XLayer', nativeCurrency: 'OKB', color: '#111111', notes: 'gas in OKB' },
  { id: 4217, key: 'tempo', name: 'Tempo', nativeCurrency: 'pathUSD', color: '#0EA5E9', notes: 'gas in pathUSD' },
  { id: 11155111, key: 'sepolia', name: 'Sepolia', nativeCurrency: 'ETH', color: '#CFB53B', notes: 'mandatory validation chain', testnet: true },
];

export const CHAIN_BY_ID: Record<number, ChainInfo> = Object.fromEntries(
  HOOKSWAP_CHAINS.map((c) => [c.id, c]),
);

export const CHAIN_BY_KEY: Record<string, ChainInfo> = Object.fromEntries(
  HOOKSWAP_CHAINS.map((c) => [c.key, c]),
);

export const PROTOCOLS = ['Uniswap v2', 'Uniswap v3', 'Universal Router', 'Permit2', 'HookSwap Launchpad'] as const;

export const KNOWLEDGE_TYPES = [
  'documentation',
  'whitepaper',
  'contract-abi',
  'audit',
  'governance',
  'pool-data',
  'token-metadata',
  'faq',
  'tutorial',
] as const;

export type ProtocolFilter = (typeof PROTOCOLS)[number];
export type KnowledgeType = (typeof KNOWLEDGE_TYPES)[number];

File: ai-platform/backend/app/core/chains.py
"""HookSwap chain registry.

Registry + adapter pattern (README §"Chain registry"): the platform is only
"live" on the chains below, but new chains are added by config alone. This
mirrors the DEX's ``contracts/deployments/*.json`` and the shared
``packages/chains/chains.ts``. Retrieval + live-data tools are chain-aware and
resolve chains through this single source of truth.
"""
from __future__ import annotations

from dataclasses import dataclass, field
from enum import IntEnum


class ChainId(IntEnum):
    ROBINHOOD = 4663
    HYPEREVM = 999
    INK = 57073
    MEGAETH = 4326
    XLAYER = 196
    TEMPO = 4217
    SEPOLIA = 11155111


@dataclass(frozen=True, slots=True)
class ChainInfo:
    chain_id: ChainId
    slug: str
    name: str
    native_symbol: str
    is_testnet: bool = False
    is_validation_chain: bool = False
    gas_token: str | None = None  # when gas is paid in a non-native ERC-20
    aliases: tuple[str, ...] = field(default_factory=tuple)


# The live registry. Order = display/priority order (Robinhood primary).
CHAINS: dict[ChainId, ChainInfo] = {
    ChainId.ROBINHOOD: ChainInfo(
        ChainId.ROBINHOOD, "robinhood", "Robinhood Chain", "ETH",
        aliases=("robinhoodchain", "rh"),
    ),
    ChainId.HYPEREVM: ChainInfo(
        ChainId.HYPEREVM, "hyperevm", "HyperEVM", "HYPE",
        aliases=("hyperliquid", "hyper"),
    ),
    ChainId.INK: ChainInfo(ChainId.INK, "ink", "Ink", "ETH"),
    ChainId.MEGAETH: ChainInfo(ChainId.MEGAETH, "megaeth", "MegaETH", "ETH"),
    ChainId.XLAYER: ChainInfo(
        ChainId.XLAYER, "xlayer", "X Layer", "OKB", gas_token="OKB",
        aliases=("okx", "oklayer"),
    ),
    ChainId.TEMPO: ChainInfo(
        ChainId.TEMPO, "tempo", "Tempo", "pathUSD", gas_token="pathUSD",
    ),
    ChainId.SEPOLIA: ChainInfo(
        ChainId.SEPOLIA, "sepolia", "Sepolia", "ETH",
        is_testnet=True, is_validation_chain=True,
    ),
}

# slug/alias -> ChainId lookup, built once.
_SLUG_INDEX: dict[str, ChainId] = {}
for _info in CHAINS.values():
    _SLUG_INDEX[_info.slug] = _info.chain_id
    _SLUG_INDEX[str(_info.chain_id.value)] = _info.chain_id
    for _alias in _info.aliases:
        _SLUG_INDEX[_alias] = _info.chain_id


def resolve_chain(value: str | int | None) -> ChainInfo | None:
    """Resolve a chain from an id, slug, or alias. Returns None if unknown."""
    if value is None:
        return None
    if isinstance(value, int) or (isinstance(value, str) and value.isdigit()):
        try:
            return CHAINS.get(ChainId(int(value)))
        except ValueError:
            return None
    cid = _SLUG_INDEX.get(str(value).strip().lower())
    return CHAINS.get(cid) if cid else None


def all_chain_ids() -> list[int]:
    return [c.value for c in CHAINS]


def is_supported(value: str | int | None) -> bool:
    return resolve_chain(value) is not None
